internal/uuid/uuid.go

```diff
@@ -20,3 +27,80 @@ func New(r io.Reader) (string, error) {
 	bs[6] = bs[6]&^0xf0 | 0x40
 	return fmt.Sprintf("%x-%x-%x-%x-%x", bs[0:4], bs[4:6], bs[6:8], bs[8:10], bs[10:]), nil
 }
+
+// Parse will use the google/uuid library to parse the string into a uuid
+// if parsing fails, it will return an empty map. It will fill the map
+// with some decoded values with fillMap
+func Parse(s string) (map[string]interface{}, error) {
+	uuid, err := uuid.Parse(s)
+	if err != nil {
+		return nil, err
+	}
+	out := make(map[string]interface{}, getVersionLen(int(uuid.Version())))
+	fillMap(out, uuid)
+	return out, nil
+}
+
+// Fills the map with values from the uuid. Version and variant for every version.
+// Version 1-2 has decodable values that could be of use, version 4 is random,
+// and version 3,5 is not feasible to extract data.
+func fillMap(m map[string]interface{}, u uuid.UUID) {
+	m["version"] = int(u.Version())
+	m["variant"] = u.Variant().String()
+	switch version := m["version"]; version {
```

Comment:
[nit] Is this needed? Couldn't it also be

```suggestion
	switch version := m["version"] {
```

Reply:
Unfortunately not :// It would have to be `switch m["version"] {` to not get a compile error. 
And then I would have to unpack it again when doing the if-statement further down. So could make a local variable instead, if you want, or I could keep it as is

Reply:
Let's keep it. Thanks for explaining this 🙃